docs/hub/adding-a-model.md

```diff
@@ -85,35 +180,123 @@ git add .
+
+![/docs/assets/hub/notebook_login.png](/docs/assets/hub/notebook_login.png)
+
+To login through the command line, run the following from a terminal:
+
+>>> huggingface-cli login
+
+You will be presented with a prompt asking for you to paste your token
+![/docs/assets/hub/cli-login.png](/docs/assets/hub/cli-login.png)
+
+After either login method is chosen, you will be asked to run:
+>>> git config --global credential.helper store
+
+This ensures that git is looking at our newly-stored credentials any time we wish to push to the Hub
+
+> Note: You may find that `HfApi` has a `set_access_token` function. This does not set all the permissions needed at each location, and is more for internal use. You should use one of the two methods mentioned above.
+
+## Creating a repository
+
+When using the `huggingface_hub`, we can create a new repository from just a few lines of code!
+First we need to instantiate the `HfApi` class, which holds all of the magic:
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+Afterwards we can run the `create_repo` function, specifying a number of settings and options for our new repository:
+>>> api.create_repo(
+>>>   name = "dummy", # The name of our repository
+>>>   organization = None, # The username of the expected repository. Automatically grabs your logged-in profile name
+>>>   private = False, # Whether the repo should be public or private
+>>>   repo_type = "model" # The type of repository, such as "model", "space", "dataset"
+>>> )
+
+> To read more about what you can pass in, check out its documentation by doing api.create_repo?
+
+## Uploading your files
+
+There are two methods for uploading a file to the Hub:
+- `HfApi.upload_file`
+- `Repository.push_to_hub`
+
+`upload_file` should be used when the file is quite small (less than 10MB), and is straightforward to use. Simply pass in the filename, the location it should be in the repository, and the name of the repository.
+
+In this example we'll write a quick `README.md` file: 
+>>> with open('README.md', 'w+') as f:
+>>>     f.write('''# Dummy model
+>>>     
+>>> This is a dummy model''')
+
+And quickly push it to the Hub:
+>>> url = api.upload_file(
+>>>    path_or_fileobj = 'README.md', 
+>>>    path_in_repo = 'README.md', 
+>>>    repo_id = 'my_username/dummy',
+>>>)
+
+You can find your file live on the Hub at the url returned from `upload_file`
+
+
+If you are trying to upload larger files to the hub (over 10MB), you should ensure that **git-lfs** is installed on your system. Git is used to manage your files on the Hub, and tracking of large file storages needs to utilize this. 
+
+* [Install Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [Install Git LFS](https://git-lfs.github.com/)
+
+You do not need to initialize `git-lfs` however, as `huggingface_hub` will automatically do this for you. 
```

Comment:
Given this should we maybe not mention the installation of them to keep the read lighter?